openfisca_core/formulas.py

```diff
@@ -774,15 +770,14 @@ def new_filled_column(
     elif isinstance(source_file_path, str):
         source_file_path = source_file_path.decode('utf-8')
 
-    if start_date is UnboundLocalError:
-        start_date = None if reference_column is None else reference_column.start
-    elif start_date is not None:
-        assert isinstance(start_date, datetime.date)
-
-    if stop_date is UnboundLocalError:
-        stop_date = None if reference_column is None else reference_column.end
-    elif stop_date is not None:
-        assert isinstance(stop_date, datetime.date)
+    if end is UnboundLocalError:
+        end = None if reference_column is None else reference_column.end
+    elif end is not None:
+        assert isinstance(end, str), 'Type error on {}. String expected. Found: {}'.format(name + '.end', type(end))
+        try:
+            get_datetime_date(end)
```

Comment:
@MattiSG,

We need another opinion to cut short this discussion.
We're inside the `new_filled_column` function, which builds an OpenFisca column (=variable) based on what has been coded by the legislation writer (`class rsa(Variable)...`).

The function `get_datetime_date ` parses a `'YYYY-MM-DD'` string and returns the equivalent `datetime.date`.

The suggested implementation here is to:

- First call the `get_datetime_date` function once when building the column, just to check if it returns an error. We do not save the result.
- [Then](https://github.com/openfisca/openfisca-core/pull/522/files#diff-31e80408104c8f43c6b796a369a77585R842), a little further, call it for _each_ dated formula to check that its start date is consistent with the variable end date.
- [Finally](https://github.com/openfisca/openfisca-core/pull/522/files#diff-31e80408104c8f43c6b796a369a77585R502), we also call it in the `find_function` method, whose job is to find the right dated function to use in a calculation. This method is called _each time_ the variable is calculated.

I think there is no reason to parse a string `1 + k + n` times.
(where`k` = numbers of dated formulas, `n` = numbers of times the variable is calculated). I'm therefore in favour of parsing this string once and for all when we build the variable.

@sandcha thinks otherwise, and I'll let her expose her arguments.

@MattiSG wdyt ?